app/models/book.rb

```diff
@@ -35,4 +35,26 @@ def times_sold
   def published_date_written
     published_date.strftime("%B %d, %Y")
   end
+
+  def apply_discount(sale)
+    self.discounted_price = price - discount_amount(sale[:discount_amount], sale[:discount_type])
+  end
+
+  def remove_discount
+    self.discounted_price = nil
+  end
+
+  def lowest_price
+    discounted_price || price
+  end
+
+  private
+
+  def discount_amount(amount, type)
+    if type == "percent"
+      price * (amount.to_f / 100)
+    else
+      amount.to_f
+    end
+  end
```

Comment:
So, your model is starting to get pretty fat. That's not necessarily bad but you'd probably be better off moving most of this logic to concerns. Annie will probably give you better insight as to _why_ and _how_. If she doesn't have time, it's fine. You can ship this.